architecture/ambient/ztunnel.md

```diff
@@ -0,0 +1,236 @@
+  * In practice, this largely means that references are fully qualified in the API. IP Addresses (generally) have a network associated with them, node names have a cluster associated with them, etc.
+
+See the [XDS Evolution](https://docs.google.com/document/d/1V5wkeBHbLSLMzAMbwFlFZNHdZPyUEspG4lHbnB0UaCg/edit) document for more history and details.
+
+The `Workload` aims to represent everything about a workload (generally a `Pod` or `WorkloadEntry`).
+This includes things like it's IP address, identity, metadata (name, namespace, app, version, etc), and whether it has a waypoint proxy associated.
+
+The `Service` aims to represent everything about a service (generally a `Service` or `ServiceEntry`).
+This includes things like its IP Addresses and ports.
+
+### Authorization Type
+
+A secondary configuration consumed by Ztunnel is the [`Authorization` resource](../../pkg/workloadapi/security/authorization.proto).
+[Original Design Doc](https://docs.google.com/document/d/17mRVzXe8PS7VoligvIx52T10tOP7xPQ9noeOzoLO2cY/edit).
+
+This resource aims to represent the relatively small set of Authorization policies that Ztunnel support.
+Most notably, this is only L4 resources.
+
+Most of the API is fairly straight forward.
+However, one interesting aspect is how these policies associate with workloads.
+Istio's AuthorizationPolicy has label selectors.
+However, we intentionally do not send those as part of the Workload API, in order to keep the size low.
+
+The obvious solution to this is to put the list of selected workloads into the policy itself.
+However, this means anytime a workload changes (often), we need to update the policy.
+
+Instead, the opposite was chosen: each workload will list the policies that select it.
+This works out to be more efficient in common cases where policies change must less often than workloads.
+This only applies for selector-based policies; namespaced and global policies can be handled without needing to list them out.
+
+## Redirection
+
+As ztunnel aims to transparently encrypt and route users traffic, we need a mechanism to capture all traffic entering and leaving "mesh" pods.
+This is a security critical task: if the ztunnel can be bypassed, authorization policies can be bypassed.
+
+Redirection must meet these requirements:
+* All traffic *egressing* a pod in the mesh should be redirected to the node-local ztunnel on port 15001.
+  * It is critical that this path preserves the Service IP, if the traffic was to a Service.
+* All traffic *ingressing* a pod on port 15008 in the mesh should be redirected to the node-local ztunnel on port 15008 (more on this later).
+* All other traffic *ingressing* a pod in the mesh should be redirected to the node-local ztunnel on port 15006.
+
+TODO: fill in implementation details of how redirection is actually implemented.
+
+## HBONE
```

Comment:
This is all good information.  But since sidecars also support HBONE, should we create a separate doc and then point to that from here?